### DISCUSSION TITLE
Is there any paper and open source code for validation with UPPAAL

### DISCUSSION THREAD

**Discussion by anuli1234566777765768** [UPPAALModelChecker/UPPAAL-Meta#265] [Q&A]
Is there any paper and open source code for validation with UPPAAL？

**Comment by mikucionisaau** [comment]
There are many papers: see [publications](https://uppaal.org/documentation/#published-material) in the [Documentation section](https://uppaal.org/documentation/).
In particular: 
- [Tutorial on Uppaal](https://uppaal.org/texts/new-tutorial.pdf)
- [Uppaal SMC Tutorial](https://uppaal.org/texts/uppaal-smc-tutorial.pdf)

Validation (as in "does this model behave as I expect") is done informally using Simulator tabs in the [Uppaal GUI](https://uppaal.org/uppaal.png).

Uppaal is not open source, but several components ([parser](https://github.com/UPPAALModelChecker/utap), [Difference Bound Matrices](https://github.com/UPPAALModelChecker/UDBM), [Clock Difference Diagrams](https://github.com/UPPAALModelChecker/UCDD), [trace interpreter](https://github.com/UPPAALModelChecker/tracer)) are published as open source libraries and tools.
